arm-containerregistry/2016-06-27-preview/swagger/containerregistry.json

```diff
@@ -0,0 +1,642 @@
+      "description": "The storage account base properties.",
+      "properties": {
+        "name": {
+          "description": "Storage account name.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "RegistryCreateParameters": {
+      "description": "The parameters for creating a container registry.",
+      "required": [
+        "location",
+        "properties"
+      ],
+      "properties": {
+        "location": {
+          "description": "Resource location.",
+          "type": "string"
+        },
+        "tags": {
+          "description": "Resource tags.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "properties": {
+          "$ref": "#/definitions/RegistryPropertiesCreateParameters",
+          "description": "The parameters for creating container registry properties.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "RegistryPropertiesCreateParameters": {
+      "description": "The parameters for creating container registry properties.",
+      "required": [
+        "tenantId",
+        "storageAccount"
+      ],
+      "properties": {
+        "tenantId": {
+          "description": "The tenant id for the current user.",
+          "type": "string"
+        },
+        "adminUserEnabled": {
+          "description": "The boolean value that indicates whether admin user is enabled.",
+          "type": "boolean"
+        },
+        "storageAccount": {
+          "$ref": "#/definitions/StorageAccountProperties",
+          "description": "The storage account properties."
+        }
+      }
+    },
+    "StorageAccountProperties": {
+      "description": "The storage account properties.",
+      "required": [
+        "name",
+        "accessKey",
+        "endPointUrl"
+      ],
+      "properties": {
+        "name": {
+          "description": "Storage account name.",
+          "type": "string"
+        },
+        "accessKey": {
+          "description": "Storage account access key.",
+          "type": "string"
+        },
+        "endPointUrl": {
+          "description": "Storage account endpoint URL.",
+          "type": "string"
+        },
+        "location": {
+          "description": "Storage account location.",
+          "type": "string"
+        }
+      }
+    },
+    "Resource": {
+      "description": "Azure resource.",
+      "properties": {
+        "id": {
+          "description": "Resource Id.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "Resource type.",
+          "type": "string",
+          "readOnly": true
+        },
+        "location": {
+          "description": "Resource location.",
+          "type": "string",
+          "readOnly": true

```

Comment:
I read the documentation suggesting location to be required in Resource before making the changes and I was actually puzzled. When we create/update a registry, we use RegistryCreateParameters/RegistryUpdateParameters. The Registry definition (which inherits Resource) will only be used in response. In that sense, why do we want to make it required? We were following the same pattern from storage account and they didn't have location as a required property (https://github.com/Azure/azure-rest-api-specs/blob/master/arm-storage/2016-01-01/swagger/storage.json#L948). Futher, shouldn't location be treated in the same way as, for example, resource name, since once they are set, they cannot be changed, unless the resource is recreated? Shouldn't location be readOnly then?